tests/multidevice/multidevice.cpp

```diff
@@ -224,4 +224,155 @@ TEST_F(multidevice_ft, ensure_color_camera_is_enabled)
+    }
+
+    if (data->device)
+    {
+        k4a_device_stop_cameras(data->device);
+        k4a_device_stop_imu(data->device);
+        k4a_device_close(data->device);
+        data->device = NULL;
+    }
+
+    return result;
+}
+
+TEST_F(multidevice_ft, start_parallel)
+{
+    LOCK_HANDLE lock;
+    THREAD_HANDLE th1, th2;
+    ASSERT_NE((lock = Lock_Init()), (LOCK_HANDLE)NULL);
+    parallel_start_data_t data1 = { 0 }, data2 = { 0 };
+
+    ASSERT_EQ((uint32_t)2, k4a_device_get_installed_count());
+
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_open(0, &data1.device));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_open(1, &data2.device));
+
+    k4a_device_configuration_t config = K4A_DEVICE_CONFIG_INIT_DISABLE_ALL;
+    config.color_format = K4A_IMAGE_FORMAT_COLOR_MJPG;
+    config.color_resolution = K4A_COLOR_RESOLUTION_2160P;
+    config.depth_mode = K4A_DEPTH_MODE_NFOV_2X2BINNED;
+    config.camera_fps = K4A_FRAMES_PER_SECOND_30;
+
+    // prevent the threads from running
+    Lock(lock);
+
+    data1.lock = data2.lock = lock;
+    data2.config = data1.config = &config;
+
+    ASSERT_EQ(THREADAPI_OK, ThreadAPI_Create(&th1, parallel_start_thread, &data1));
+    ASSERT_EQ(THREADAPI_OK, ThreadAPI_Create(&th2, parallel_start_thread, &data2));
+
+    // start the test
+    Unlock(lock);
+
+    // Wait for the threads to terminate
+    int result1, result2;
+    ASSERT_EQ(THREADAPI_OK, ThreadAPI_Join(th1, &result1));
+    ASSERT_EQ(THREADAPI_OK, ThreadAPI_Join(th2, &result2));
+
+    ASSERT_EQ(result1, K4A_RESULT_SUCCEEDED);
+    ASSERT_EQ(result2, K4A_RESULT_SUCCEEDED);
+
+    if (data1.device)
+    {
+        k4a_device_close(data1.device);
+    }
+
+    if (data2.device)
+    {
+        k4a_device_close(data2.device);
+    }
+
+    Lock_Deinit(lock);
+}
+
+TEST_F(multidevice_ft, close_parallel)
+{
+    LOCK_HANDLE lock;
+    THREAD_HANDLE th1, th2;
+    ASSERT_NE((lock = Lock_Init()), (LOCK_HANDLE)NULL);
+    parallel_start_data_t data1 = { 0 }, data2 = { 0 };
+
+    ASSERT_EQ((uint32_t)2, k4a_device_get_installed_count());
+
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_open(0, &data1.device));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_open(1, &data2.device));
+
+    k4a_device_configuration_t config = K4A_DEVICE_CONFIG_INIT_DISABLE_ALL;
+    config.color_format = K4A_IMAGE_FORMAT_COLOR_MJPG;
+    config.color_resolution = K4A_COLOR_RESOLUTION_2160P;
+    config.depth_mode = K4A_DEPTH_MODE_NFOV_2X2BINNED;
+    config.camera_fps = K4A_FRAMES_PER_SECOND_30;
+
+    data2.config = data1.config = &config;
+    data1.lock = data2.lock = lock;
+
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_cameras(data1.device, data1.config));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_cameras(data2.device, data2.config));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_imu(data1.device));
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_start_imu(data2.device));
+
+    data1.started = data2.started = true;
```

Comment:
Were parallel closes causing an issue before? It seems like we weren't closing the dll handle before and the engine destroy flow shouldn't really be affected.

Reply:
No, parallel closes were not an issue before. And to your point we were not fully closing the handle so closes were unknown. This test establishes that we don't have issues with parallel closes.